consult.el

```diff
@@ -2149,5 +2149,87 @@ if not available use `default-directory'."
 
 (add-hook 'consult--completion-refresh-hook #'consult--icomplete-refresh)
 
+
+;;;; find, fd, locate
+(defun consult--guess-initial-dir (&optional askdir)
+  "Returns root directory for use in `consult' searching functions.
+If ASKDIR is not `nil' prompt the user for the inital directory,
+else use simple logic: first try `project', then `projectile',
+and defaults to `default-directory'."
+  (expand-file-name
+   (if askdir
+       (read-directory-name "Initial directory: ")
```

Comment:
And the askdir functionality should go into `consult-grep-directory-default`.

Reply:
Shouldn't be better if the function mentioned previously include the asking part?

Reply:
We could also refactor this consult-directory-default and consult--with-directory and move all the logic into consult-directory-default if you think that's better.